### PR TITLE
Rename _isPlaying to _playbackStarted. 

### PR DESCRIPTION
**Pull request by bclwhitaker**
Revert removal of _isPlaying check, so that pause will not trigger if the video has ended and pause() is called.  Still make sure that we capture the paused state before we call pause though.

### CODE REVIEW

**Comment by gkatsev** [comment]
LGTM.

File: src/com/videojs/providers/HTTPVideoProvider.as [hunk 1]
```diff
@@ -291,7 +291,7 @@ package com.videojs.providers{
         public function pause():void{
             var alreadyPaused = _isPaused;
             _ns.pause();
-            if(!alreadyPaused){
+            if(_playbackStarted && !alreadyPaused){
```
Comment (gkatsev) [inline src/com/videojs/providers/HTTPVideoProvider.as]: this is the important bit.

File: src/com/videojs/providers/HTTPVideoProvider.as [hunk 2]
```diff
@@ -213,7 +213,7 @@ package com.videojs.providers{
         }
         
         public function get playing():Boolean{
-            return _isPlaying;
+            return _playbackStarted;
```
Comment (heff) [inline src/com/videojs/providers/HTTPVideoProvider.as]: playbackStarted to me sounds like the video has started playing but may currently be paused or playing. Is that what this is meant to be?

Comment (bclwhitaker) [inline src/com/videojs/providers/HTTPVideoProvider.as]: Yea, that's it exactly and why I thought the original name was misleading...

Comment (gkatsev) [inline src/com/videojs/providers/HTTPVideoProvider.as]: yes, from what @bclwhitaker and I understood, this is what this variable actually means. It's true from when a user as pressed play/seeked and until the video has ended.

**Comment by mmcc** [comment]
:+1:

File: src/com/videojs/providers/HTTPVideoProvider.as (hunk 2 above)
Comment (heff) [inline src/com/videojs/providers/HTTPVideoProvider.as]: Got it. The `playing()` function is pretty poorly named too then. Thanks for clearing that up.

**Comment by heff** [comment]
Looks good to me. Could this affect other ad plugins some how?

**Comment by gkatsev** [comment]
The new version of the ad plugin needs these changes.

File: src/com/videojs/providers/HTTPVideoProvider.as (hunk 2 above)
Comment (heff) [inline src/com/videojs/providers/HTTPVideoProvider.as]: @bclwhitaker yeah `_isPlaying && _isPaused` is ridiculous. In core we have the `vjs-has-started` class if you're looking for similar naming, though I've considered changing that to `vjs-played` to match `vjs-ended` and the video element's `played` property. What you have here is good though.

**Comment by bclwhitaker** [comment]
Ad plugins using the current version of contrib-ads should be positively impacted as well with this change (and Gary's which we pulled in earlier).  It takes care of an issue with preroll behavior and autoplay settings which were causing the swf to report the paused state incorrectly.

**Pull request merged**